src/internal/ScrollCaptor.js

```diff
@@ -18,41 +18,67 @@ class ScrollCaptor extends Component<CaptorProps> {
   scrollTarget: HTMLElement;
   touchStart: number;
 
+  // Scroll element previous state variables
+  previousScrollTop: number = 0;
+  previousScrollHeight: number = 0;
+  previousClientHeight: number = 0;
+
+  // Event listener flags
+  listeningToScroll: boolean = false;
+  listeningToTouchStart: boolean = false;
+  listeningToTouchMove: boolean = false;
+
   componentDidMount() {
-    this.startListening(this.scrollTarget);
+    this.previousScrollHeight = this.scrollTarget.scrollHeight;
+    this.previousClientHeight = this.scrollTarget.previousClientHeight;
+    this.refreshListening(this.scrollTarget);
+  }
+
+  componentDidUpdate() {
+    if (this.previousScrollHeight !== this.scrollTarget.scrollHeight || this.previousClientHeight !== this.scrollTarget.clientHeight) {
+      this.previousScrollHeight = this.scrollTarget.scrollHeight;
+      this.previousClientHeight = this.scrollTarget.clientHeight;
+      this.refreshListening(this.scrollTarget);
+    }
   }
+
   componentWillUnmount() {
     this.stopListening(this.scrollTarget);
   }
-  startListening(el: HTMLElement) {
+  refreshListening(el: HTMLElement) {
     // bail early if no scroll available
```

Comment:
maybe we should also bail early if we are already listening to everything

Reply:
Yep, done.